src/libraries/System.Net.HttpListener/tests/SimpleHttpTests.cs

```diff
@@ -169,7 +169,6 @@ public async Task UnknownHeaders_Success(int numHeaders)
         }
 
         [Fact]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/30284")]
         public void ListenerRestart_BeginGetContext_Success()
```

Comment:
Can you expand this test to give the same level of coverage as the sync test below?

However, the catch block around Start() should be more discriminating about what exceptions it considers a port conflict.

Reply:
> Can you expand this test to give the same level of coverage as the sync test below?

Changed - test is now shared for both `GetContext` and `BeginGetContext`.

> the catch block around Start() should be more discriminating about what exceptions it considers a port conflict

I changed the cought exception to `HttpListenerException`, but it's hard to narrow it down more. It can differ based on platform and based on whether the port is taken by some other socket or another `HttpListener` instance.